clang/test/Analysis/setgid-setuid-order.c

```diff
@@ -0,0 +1,170 @@
+// RUN: %clang_analyze_cc1 -analyzer-checker=core,security.SetgidSetuidOrder -verify %s
+
+#include "Inputs/system-header-simulator-setgid-setuid.h"
+
+void correct_order() {
+  if (setgid(getgid()) == -1)
+    return;
+  if (setuid(getuid()) == -1)
+    return;
+  if (setgid(getgid()) == -1)
+    return;
```

Comment:
Here is a sequence of `setuid(getuid())` and `setgid(getgid())` in the code, but still this should be not recognized as error.

Reply:
But why do the SEI-CERT best practices mandate that "this should not be recognized as an error"? Could you briefly explain this in a comment? (E.g. "Special case: calling `setgid(getgid())` after an earlier `setgid(getgid()); setuid(getuid())` combination is legitimate, because it... <why is this good>")

Reply:
I also find it weird that we start the test file with an edge-case :D